examples/researcher/researcher.ipynb

```diff
@@ -0,0 +1,605 @@
+   "execution_count": 1,
+   "id": "1e9e0882-d0ed-4c19-a404-f0957ef2b677",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "// Deno imports\n",
+    "import Metaphor from 'npm:metaphor-node';\n",
+    "import OpenAI from 'npm:openai';\n",
+    "\n",
+    "// NodeJS imports\n",
+    "//import Metaphor from 'metaphor-node';\n",
+    "//import OpenAI from 'openai';\n",
+    "\n",
+    "\n",
+    "const METAPHOR_API_KEY = // insert or load your API key here\n",
+    "const OPENAI_API_KEY = // insert or load your API key here\n",
+    "\n",
+    "const metaphor = new Metaphor(METAPHOR_API_KEY);\n",
+    "const openai = new OpenAI({ apiKey: OPENAI_API_KEY });"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "af45bf85-f40d-4f27-a697-083c1fc37b08",
+   "metadata": {},
+   "source": [
+    "Since we'll be making several calls to the OpenAI API to get a completion from GPT 3.5-turbo, let's make a simple utility wrapper function so we can pass in the system and user messages directly, and get the LLM's response back as a string."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "id": "1ad7dbb0-3732-4039-a818-fb74c79fe7f9",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "async function getLLMResponse({system = 'You are a helpful assistant.', user = '', temperature = 1, model = 'gpt-3.5-turbo'}){\n",
+    "    const completion = await openai.chat.completions.create({\n",
+    "        model,\n",
+    "        temperature,\n",
+    "        messages: [\n",
+    "            {'role': 'system', 'content': system},\n",
+    "            {'role': 'user', 'content': user},\n",
+    "        ]\n",
+    "    });\n",
+    "    return completion.choices[0].message.content;\n",
+    "}"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "4b15f155-fbde-42da-907a-19328321acf6",
+   "metadata": {},
+   "source": [
+    "Okay, great! Now let's starting building Metaphor Researcher. The app should be able to automatically generate research reports for all kinds of different topics. Here's two to start:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "id": "811ea07b-c9c4-42a5-bbf2-62f162aca734",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "const XYZZY_TOPIC = 'xyzzy';\n",
+    "const ART_TOPIC = 'renaissance art'; "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "344c7cd8-577c-4eb4-870e-9ed813093d43",
+   "metadata": {},
+   "source": [
+    "The first thing our app has to do is decide what kind of search to do for the given topic. \n",
+    "\n",
+    "Metaphor offers two kinds of search: **neural** and **keyword** search. Here's how we decide:\n",
+    "\n",
+    "- Neural search is preferred because lets us retrieve high quality, semantically relevant data. It is especially suitable when a topic is well-known and popularly discussed on the Internet, allowing the machine learning model to retrieve contents which are more likely recommended by real humans.  \n",
```

Comment:
```suggestion
    "- Neural search is preferred because it lets us retrieve high quality, semantically relevant data. It is especially suitable when a topic is well-known and popularly discussed on the Internet, allowing the machine learning model to retrieve contents which are more likely recommended by real humans.  \n",
```